include/aspect/mesh_deformation/fastscape.h

```diff
@@ -0,0 +1,680 @@
+     * ASPECT output folder.
+     */
+    void fastscape_named_vtk_(double *fp, const double *vexp, int *astep, const char *c, int *length);
+
+    /**
+     * Copy the current FastScape topography.
+     */
+    void fastscape_copy_h_(double *hp);
+
+    /**
+     * Copy the current FastScape basement.
+     */
+    void fastscape_copy_basement_(double *b);
+
+    /**
+     * Copy the current FastScape silt fraction.
+     */
+    void fastscape_copy_f_(double *sf);
+
+    /**
+     * Initialize the straigraphy component of FastScape.
+     */
+    void fastscape_strati_(const int *nstepp, const int *nreflectorp, int *nfreqp, const double *vexp);
+
+    /**
+     * Copy the current FastScape slopes.
+     */
+    void fastscape_copy_slope_(double *slopep);
+
+    /**
+     * Destroy FastScape.
+     */
+    void fastscape_destroy_();
+#ifdef  __cplusplus
+  }
+#endif
+
+  /**
+   * A plugin that utilizes the landscape evolution code FastScape
+   * to deform the ASPECT boundary through advection, uplift,
+   * hillslope diffusion, the stream power law, sediment deposition
+   * and marine diffusion.
+   */
+  template<int dim>
+  class FastScape : public Interface<dim>, public SimulatorAccess<dim>
+  {
+    public:
+      /**
+       * Initialize variables for FastScape.
+       */
+      virtual void initialize ();
+
+      /**
+        * A function that creates constraints for the velocity of certain mesh
+        * vertices (e.g. the surface vertices) for a specific boundary.
+        * The calling class will respect
+        * these constraints when computing the new vertex positions.
+        */
+      virtual
+      void
+      compute_velocity_constraints_on_boundary(const DoFHandler<dim> &mesh_deformation_dof_handler,
+                                               AffineConstraints<double> &mesh_velocity_constraints,
+                                               const std::set<types::boundary_id> &boundary_id) const;
+
+      /**
+       * Declare parameters for the FastScape plugin.
+       */
+      static
+      void declare_parameters (ParameterHandler &prm);
+
+      /**
+       * Parse parameters for the FastScape plugin.
+       */
+      void parse_parameters (ParameterHandler &prm);
+
+      /**
+       * Function used to set the FastScape ghost nodes. FastScape boundaries are
+       * not uplifted or periodic for advection and diffusion. By using a layer
+       * of extra nodes in the FastScape model, we can avoid seeing FastScape boundary
+       * effects within ASPECT. Similalrly, we can use these nodes to have fully
+       * periodic boundaries, where we check the flow direction and update the FastScape
+       * ghost nodes, and the nodes one layer inward (boundary nodes in ASPECT) to match
+       * the parameters on the other side (vx, vy, vz, h). This is done every ASPECT timestep
+       * before running FastScape.
+       * TODO: Should this be in private?
```

Comment:
Yes, if this function is never called from outside this class, it should be private. It is good practice to hide all unnecessary functions from outside access (because it makes it easier for you to figure out later, where this function could be called from, and reduces the potential for tight coupling between different  code pieces).

Reply:
Good to know! In that case I've moved all the new functions into private.